### ISSUE TITLE
TypeError: ast.remove is not a function

### ISSUE THREAD

**Issue by thenewvu** [csscomb/csscomb.js#501]
Sir, I've met this error with all CSS files and with both custom and default configure:

```
(node:25107) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: ast.remove is not a function
(node:25107) DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

This error only occurs in v4.0.1, not with v3.1.7.

My environment:

```
node v7.7.1
arch linux 4.9.11
```

**Comment by sublistic** [comment]
I get the following similar error:

    (node:2323) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: Cannot read property 'is' of null
    (node:2323) DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

node v7.10.0
OSX 10.11.6

**Comment by lclghst** [comment]
Same error here

```
node 7.10.0
OSX 10.10.5
```

**Comment by gabriensymons** [comment]
Similar error when runing csscomb@4.0.1 as an exec command with ant. Like @thenewvu, no error with version 3.1.7:
```csscomb:
     [exec] (node:66140) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: this._parseString(...).then is not a function```

My env:
```
Apache Ant 1.9.7
node 6.10.2
macOS 10.12.4
```

**Comment by janwerkhoven** [comment]
Getting error in Atom Beautify 0.29.26. How to resolve?

```
TypeError: ast.remove is not a function
    at /Users/jw/.atom/packages/atom-beautify/node_modules/csscomb/lib/options/block-indent.js:48:13
    at Node.eachFor (/Users/jw/.atom/packages/atom-beautify/node_modules/gonzales-pe/lib/gonzales.js:129:83)
    at Object.process (/Users/jw/.atom/packages/atom-beautify/node_modules/csscomb/lib/options/block-indent.js:44:9)
    at /Users/jw/.atom/packages/atom-beautify/node_modules/csscomb/lib/core.js:380:14
    at Array.forEach (native)
    at Comb._processTree (/Users/jw/.atom/packages/atom-beautify/node_modules/csscomb/lib/core.js:379:8)
```

**Comment by janwerkhoven** [comment]
Temporary workaround: Remove `"block-indent": "  ",` from the `.csscomb.json` file and the error ceases to appear and block the rest of the formatting. That means no indenting of blocks until fixed, but still better than no formatting at all.

**Comment by janwerkhoven** [comment]
Is it this line of code that is failing?
https://github.com/csscomb/csscomb.js/blob/dev/src/options/block-indent.js#L48

**Comment by janwerkhoven** [comment]
Is nobody maintaining this?

**Comment by aiboy** [comment]
@janwerkhoven I do maintain this repository. I will look into this asap.

**Comment by mfeckie** [comment]
Should this

https://github.com/csscomb/csscomb.js/blob/dev/src/options/block-indent.js#L48

`ast.remove(i);`

not be

`ast.removeChild(i);` ?

**Comment by ryanbowden** [comment]
I am having the same issue and changing 
`"eof-newline": false,` 
to true fixes the fault. Hope that helps finding the issue.

**Comment by elmorec** [comment]
@ryanbowden It works! Thanks.

**Comment by Spunkie** [comment]
The fix from @mfeckie worked for me.

**Comment by maxfenton** [comment]
Still having this issue in the current version until I make the change @ryanbowden suggested.

**Comment by tucq88** [comment]
Still having issue, @ryanbowden update does'nt work with me :(

**Comment by tucq88** [comment]
Found the issues, cause my style has many lines commented out by `//` , Is this a known issue ?

**Comment by janwerkhoven** [comment]
I can confirm both the fix and work-around.

### How to avoid the errors
In the `.csscomb.json` set `"eof-newline": true,`.

Thanks @ryanbowden 

### How to fix it

In `node_modules/csscomb/lib/options/block-indent.js` replace `ast.remove(i);` with `ast.removeChild(i);`.

Thanks @mfeckie !

**Comment by seezee** [comment]
Also getting this error. Node v.8.9.1 & OS High Sierra with latest CSSComb.

**Comment by harapeko** [comment]
similar error

```scss
.hoge {
    $arr: (
    "s": 3,
    "m": 4,
    "l": 5,
    ) !default;

    @each $key, $value in $arr {
        "&--of#{$value} &-unit" {
            width: (100% / #{$value});
        }
    }
}
```

```
(node:110565) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): [object Object]
(node:110565) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Node: v8.7.0
OS: Ubuntu 14.04.5 LTS

**Comment by klavatron** [comment]
Thanks, both fixes work!

**Comment by xianshenglu** [comment]
@ryanbowden ,It works for me too!

**Comment by MrFebro** [comment]
*{
  outline: none;
}*****
My issue was this last asteristic which was preventing my code to run. Check if you don't have such syntax error because I have tried all the solutions above, but ended up to be that!

**Comment by tonyganch** [comment]
Fixed by @janwerkhoven with #552.

**Comment by janwerkhoven** [comment]
🙌
